src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -1687,173 +1688,41 @@ internal static void VarDecFromR4(float input, out DecCalc result)
             /// <summary>
             /// Convert double to Decimal
             /// </summary>
+
             internal static void VarDecFromR8(double input, out DecCalc result)
             {
                 result = default;
 
-                // The most we can scale by is 10^28, which is just slightly more
-                // than 2^93.  So a float with an exponent of -94 could just
+                // The smallest non-zero decimal we can represent is 10^-28, which is just slightly more
+                // than 2^-93.  So a float with an exponent of -94 could just
                 // barely reach 0.5, but smaller exponents will always round to zero.
+                // This is a shortcut to catch doubles that are too small efficiently.
                 //
-                const uint DBLBIAS = 1022;
```

Comment:
As mentioned, this is conceptually incorrect, as possibly even logically incorrect